x/mongo/driver/topology/fsm.go

```diff
@@ -228,30 +228,75 @@ func (f *fsm) checkIfHasPrimary() {
 	}
 }
 
-func (f *fsm) updateRSFromPrimary(s description.Server) {
-	if f.SetName == "" {
-		f.SetName = s.SetName
-	} else if f.SetName != s.SetName {
-		f.removeServerByAddr(s.Addr)
-		f.checkIfHasPrimary()
-		return
+// hasStalePrimary returns true if the topology has a primary that is stale.
+func hasStalePrimary(fsm fsm, srv description.Server) bool {
+	if srv.WireVersion == nil {
+		return true
+	}
+
+	// Compare the election ID values of the server and the topology lexicographically.
+	compRes := bytes.Compare(srv.ElectionID[:], fsm.maxElectionID[:])
+
+	if srv.WireVersion.Max < 17 {
+		return compRes == -1 || fsm.maxSetVersion > srv.SetVersion
```

Comment:
```suggestion
		// Pre 6.0, return true (stale primary) if setVersion is less than maxSetVersion or electionID is
		// less than maxElectionID. 
		return srv.Version < fsm.maxSetVersion || compRes == -1
```
The ordering of the conditional here is a little hard for me to parse. From what I understand, pre 6.0 we want to compare the set version of the server and the max set version seen so far. If those are equal or the set version of the server is less than the max set version seen so far, we want to compare the election ID to the max election ID seen so far. Can we leave a comment above as I've suggested and change the ordering of the conditional so it reads in the same order as the logic (and the reversal of tuple order is more clear 6.0+ below)?

Reply:
I can definitely add a comment! 

I'm not sure I follow why the ordering of the condition matters, currently it should be the same order (albeit the inverse) as the logic [here](https://github.com/mongodb/specifications/commit/37c32cf5200cd8da6652c75ac65279c43628fe7a#diff-52a48d3b84e9e90e30fb87b3799c6302750f1ed2817b2abbead44e7a7bb6bb47R1117).